lang/java/reef-common/src/main/java/org/apache/reef/driver/restart/DriverRestartManager.java

```diff
@@ -107,7 +108,16 @@ public synchronized void onRestart() {
 
     driverRuntimeRestartManager.informAboutEvaluatorFailures(getFailedEvaluators());
 
-    // TODO[REEF-560]: Call onDriverRestartCompleted() on a Timer.
+    if (driverRestartEvaluatorRecoveryMillis != Integer.MAX_VALUE) {
+      // Don't use Clock here because if there is an event scheduled, the driver will not be idle, even if
+      // driver restart has already completed, and we cannot cancel the event.
+      new Timer().schedule(new TimerTask() {
+        @Override
+        public void run() {
+          onDriverRestartCompleted();
```

Comment:
@markusweimer 
1. An evaluator checkin after this time is ignored.
2. That sounds like a good idea. Do you think it should be another item instead? Or should I send another update for that? Orthogonal to this PR, I think there are a few more ideas worth considering: Should we report the Evaluators to expect in ``DriverRestartedHandler``, and should we have a special ``EvaluatorFailedHandler`` for failures on Driver restart s.t. the user can know the difference?
3. This will only fire once. It is a synchronized call with a boolean check, though we can potentially cancel the Timer event after ``onDriverRestartCompleted`` is called.